lib/chef/resource.rb

```diff
@@ -352,6 +353,10 @@ def subscribes(action, resources, timing = :delayed)
       true
     end
 
+    # Helper for Subscribes
+    def validate_resource_spec_sub!(resources)
+      run_context.resource_collection.validate_lookup_spec!(resources)
+    end
```

Comment:
This PR is fine but we don't need another helper and can just call `validate_resource_spec` that notifies uses.